extensions/amp-story/1.0/amp-story-reaction-quiz.js

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpStoryReaction, ReactionType} from './amp-story-reaction';
+import {CSS} from '../../../build/amp-story-reaction-quiz-1.0.css';
+import {LocalizedStringId} from '../../../src/localized-strings';
+import {Services} from '../../../src/services';
+import {createShadowRootWithStyle} from './utils';
+import {dev} from '../../../src/log';
+import {htmlFor} from '../../../src/static-template';
+import {toArray} from '../../../src/types';
+
+/** @const {string} */
+const TAG = 'amp-story-reaction-quiz';
+
+/**
+ * Generates the template for the quiz.
+ *
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildQuizTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <div
+      class="i-amphtml-story-reaction-quiz-container i-amphtml-story-reaction-container"
```

Comment:
The ...reaction-content class marks that the abstract class will search for options in the component, and add the post-selection class after interacting.